src/gate_detect/src/GateDetectionNode.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+ * Created By: Cameron Newton
+ * Created On: February 27th, 2018
+ * Set up environment for GateDetectionNode
+ */
+
+#include "GateDetectionNode.h"
+
+GateDetectionNode::GateDetectionNode(int argc, char** argv) {
+    ros::NodeHandle nh;
+    ros::NodeHandle nh_;
+    image_transport::ImageTransport it(nh);
+
+    dynamic_reconfigure::Server<gate_detect::gatedetectConfig> server;
+    dynamic_reconfigure::Server<gate_detect::gatedetectConfig>::CallbackType f;
+
+    subscribeTopic = "/vision/output";
+    publishTopic   = "/gateDetect/output";
+
+    subscriber_ = it.subscribe(
+    subscribeTopic, 2, &GateDetectionNode::subscriberCallBack, this);
+
+    gateDetection = GateDetection();
+
+    f = boost::bind(&GateDetectionNode::reconfigCallBack, this, _1, _2);
+    server.setCallback(f);
+
+    publisher1_ = nh_.advertise<gate_detect::GateDetectMsg>(publishTopic, 10);
+    publisher2_ = it.advertise("gate_image_output", 100);
+
+    ros::spin();
+}
+
+void GateDetectionNode::subscriberCallBack(
+const sensor_msgs::ImageConstPtr& msg) {
+    cv_bridge::CvImagePtr cv_ptr;
+
+    try {
+        cv_ptr = cv_bridge::toCvCopy(msg, sensor_msgs::image_encodings::MONO8);
+    } catch (cv_bridge::Exception& e) {
+        ROS_ERROR("cv_bridge exception: %s", e.what());
+        return;
+    }
+
+    lineImg = cv_ptr->image;
+
+    std::vector<float> gateVector = gateDetection.initialize(lineImg);
+
+    publishGateDetectMsg(gateVector);
+
+    // Uncomment if you want to view gate seen by node
+    // publishGateImage(gateVector);
```

Comment:
We should make this conditional on a Rosparam, ie:
```
if (view_gate){
    publishGateImage(gateVector)
}
```
(where `view_gate` is set by a rosParam) 

This makes it really easy to turn it on/off without having to recompile the code, and also removes commented out code.